libcaf_net/test/doorman.cpp

```diff
@@ -0,0 +1,174 @@
+#include "caf/net/tcp_accept_socket.hpp"
+#include "caf/uri.hpp"
+
+#include "caf/test/dsl.hpp"
+
+#include "caf/net/test/host_fixture.hpp"
+
+using namespace caf;
+using namespace caf::net;
+using namespace std::literals::string_literals;
+
+namespace {
+
+struct fixture : test_coordinator_fixture<>, host_fixture {
+  fixture() {
+    mpx = std::make_shared<multiplexer>();
+    if (auto err = mpx->init())
+      CAF_FAIL("mpx->init failed: " << sys.render(err));
+    mpx->set_thread_id();
+    CAF_CHECK_EQUAL(mpx->num_socket_managers(), 1u);
+    auth.port = 0;
+    auth.host = "0.0.0.0"s;
+  }
+
+  bool handle_io_event() override {
+    return mpx->poll_once(false);
+  }
+
+  multiplexer_ptr mpx;
+  uri::authority_type auth;
+};
+
+class dummy_application {
+public:
+  static expected<std::vector<byte>> serialize(actor_system& sys,
+                                               const type_erased_tuple& x) {
+    std::vector<byte> result;
+    binary_serializer sink{sys, result};
+    if (auto err = message::save(sink, x))
+      return err.value();
+    return result;
+  }
+
+  template <class Parent>
+  error init(Parent&) {
+    return none;
+  }
+
+  template <class Parent>
+  void write_message(Parent& parent,
+                     std::unique_ptr<endpoint_manager_queue::message> msg) {
+    parent.write_packet(msg->payload);
+  }
+
+  template <class Parent>
+  error handle_data(Parent&, span<const byte>) {
+    return none;
+  }
+
+  template <class Parent>
+  void resolve(Parent&, string_view path, const actor& listener) {
+    anon_send(listener, resolve_atom::value,
+              "the resolved path is still "
+                + std::string(path.begin(), path.end()));
+  }
+
+  template <class Parent>
+  void timeout(Parent&, atom_value, uint64_t) {
+    // nop
+  }
+
+  template <class Parent>
+  void new_proxy(Parent&, actor_id) {
+    // nop
+  }
+
+  template <class Parent>
+  void local_actor_down(Parent&, actor_id, error) {
+    // nop
+  }
+
+  void handle_error(sec) {
+    // nop
+  }
+};
+
+class dummy_application_factory {
+public:
+  using application_type = dummy_application;
+
+  static expected<std::vector<byte>> serialize(actor_system& sys,
+                                               const type_erased_tuple& x) {
+    return dummy_application::serialize(sys, x);
+  }
+
+  template <class Parent>
+  error init(Parent&) {
+    return none;
+  }
+
+  application_type make() const {
+    return dummy_application{};
+  }
+};
+
+} // namespace
+
+CAF_TEST_FIXTURE_SCOPE(doorman_tests, fixture)
+
+CAF_TEST(tcp connect) {
+  auto acceptor = unbox(make_tcp_accept_socket(auth, false));
+  auto port = unbox(local_port(socket_cast<network_socket>(acceptor)));
+  auto acceptor_guard = make_socket_guard(acceptor);
+  CAF_MESSAGE("opened acceptor on port " << port);
+  uri::authority_type dst;
+  dst.port = port;
+  dst.host = "localhost"s;
+  auto conn = make_socket_guard(unbox(make_connected_tcp_stream_socket(dst)));
```

Comment:
Is there really no better way to test this than making localhost connections?

Reply:
I actually don't think there is. Do you think one could try to use unix sockets to test such a routine?

Reply:
I have split up the two test suites to cleanup the test scaffold. 
I also discussed the doorman test with @josephnoir and we came to the conclusion that:
1. splitting up the tests would be a good idea.
2. Testing tcp accept functionality without actual tcp would not be a good idea.
3. The doorman test could be realized without a localhost connection, but would be pretty hacky.. (correct me if I'm wrong Joseph.).